pilot/pkg/xds/ads.go

```diff
@@ -596,6 +596,10 @@ func (s *DiscoveryServer) initProxyMetadata(node *core.Node) (*model.Proxy, erro
 	// Update the config namespace associated with this proxy
 	proxy.ConfigNamespace = model.GetProxyConfigNamespace(proxy)
 	proxy.XdsNode = node
+	// Update the network of proxy to default system namespace's network if it is not set.
+	if proxy.Metadata.Network == "" && s.Env.NetworkManager != nil {
+		proxy.Metadata.Network = s.Env.NetworkManager.DefaultNetworkForCluster(proxy.Metadata.ClusterID)
```

Comment:
initProxyMetadata will be called only once per connection, so if namespace network update, it can not take effect

Reply:
Revised this logic in `computeProxyState`